Some improvements

sjaeckel opened the pull request:
Some more improvements on top of #677 

@gregoa could you please check if this fixes the x32 build I've seen happening on Debian?

sjaeckel committed:
Re-work some of the demos

* Add at least a `-h` option, sometimes more.
* Fix the exit codes of some of them.
* Make them all compile with `LTC_EASY`.

Signed-off-by: Steffen Jaeckel <s@jaeckel.eu>

sjaeckel committed:
Introduce `CRYPT_ERR_NUM` and build-time check `err_2_str[]` size

Signed-off-by: Steffen Jaeckel <s@jaeckel.eu>

sjaeckel committed:
Fix build on x32

* `CONSTPTR()` must be 32bit.
* `CRYPT_HASH_OVERFLOW` test operates on `unsigned long`, which is only
  32bit wide on x32.

Signed-off-by: Steffen Jaeckel <s@jaeckel.eu>

sjaeckel committed:
There are no broken demos anymore

... at least at compile time.

Signed-off-by: Steffen Jaeckel <s@jaeckel.eu>

levitte commented:
I added a fixup patch to give the compiler a chance to warn if `err_2_str` gets over-filled.

@sjaeckel, you know how to handle such commits, yeah?  `git rebase -i --autosquash`?

sjaeckel committed:
Handle potential pkg-config failures

There are cases where we want to use an MPI provider, but the package is
not installed in the system, so `pkg-config` can't find it.
To solve this simply ignore the errors created and only print something
if the user asks us to by passing `V=1`.

Signed-off-by: Steffen Jaeckel <s@jaeckel.eu>

sjaeckel committed:
Remove even more footers

Fixes 2400883060501eed9858b4fe4a12d9552635418f

Signed-off-by: Steffen Jaeckel <s@jaeckel.eu>

sjaeckel commented:
> I added a fixup patch to give the compiler a chance to warn if `err_2_str` gets over-filled.
> 
> @sjaeckel, you know how to handle such commits, yeah? `git rebase -i --autosquash`?

Thanks!

I've added one more patch with what resulted in the build failures of the previous PR, + one more clean-up patch.

levitte reviewed (approved):
Saw what you did.  Looks fine by me
